js/apps/admin-ui/src/realm-settings/user-profile/AttributesGroupForm.tsx

```diff
@@ -110,82 +110,63 @@ export default function AttributesGroupForm() {
+            isHorizontal
+            role="manage-realm"
+            onSubmit={handleSubmit(onSubmit)}
           >
-            <KeycloakTextInput
-              id="kc-name"
-              isReadOnly={!!matchingGroup}
-              {...form.register("name", { required: true })}
+            <TextControl
+              name="name"
+              label={t("nameField")}
+              labelIcon={t("nameHintHelp")}
+              rules={{ required: t("required") }}
+              readOnlyVariant={matchingGroup ? "default" : undefined}
             />
-            {!!matchingGroup && (
-              <input type="hidden" {...form.register("name")} />
-            )}
-          </FormGroup>
-          <FormGroup
-            label={t("displayHeaderField")}
-            fieldId="kc-display-header"
-            labelIcon={
-              <HelpItem
-                helpText={t("displayHeaderHintHelp")}
-                fieldLabelId="displayHeaderField"
-              />
-            }
-          >
-            <KeycloakTextInput
-              id="kc-display-header"
-              {...form.register("displayHeader")}
+            <TextControl
+              name="displayHeader"
+              label={t("displayHeaderField")}
+              labelIcon={t("displayHeaderHintHelp")}
             />
-          </FormGroup>
-          <FormGroup
-            label={t("displayDescriptionField")}
-            fieldId="kc-display-description"
-            labelIcon={
-              <HelpItem
-                helpText={t("displayDescriptionHintHelp")}
-                fieldLabelId="displayDescriptionField"
-              />
-            }
-          >
-            <KeycloakTextInput
-              id="kc-display-description"
-              {...form.register("displayDescription")}
+            <TextControl
+              name="displayDescription"
+              label={t("displayDescriptionField")}
+              labelIcon={t("displayDescriptionHintHelp")}
             />
-          </FormGroup>
-          <TextContent>
-            <Text component="h2">{t("annotationsText")}</Text>
-          </TextContent>
-          <FormGroup label={t("annotationsText")} fieldId="kc-annotations">
-            <FormProvider {...form}>
-              <KeyValueInput label={t("annotationsText")} name="annotations" />
-            </FormProvider>
-          </FormGroup>
-          <ActionGroup>
-            <Button variant="primary" type="submit" data-testid="saveGroupBtn">
-              {t("save")}
-            </Button>
-            <Button
-              variant="link"
-              component={(props) => (
-                <Link
-                  {...props}
-                  to={toUserProfile({ realm, tab: "attributes-group" })}
+            <TextContent>
+              <Text component="h2">{t("annotationsText")}</Text>
+            </TextContent>
+            <FormGroup label={t("annotationsText")} fieldId="kc-annotations">
+              <FormProvider {...form}>
```

Comment:
There should be no need for this `FormProvider`, as we already have one higher up.

Reply:
missed it, ty - removed